### ISSUE TITLE
`cargo test --doc` does not test examples

### ISSUE THREAD

**Issue by mkroman** [rust-lang/cargo#2684]
When running `cargo doc --test` in a project folder, cargo does not test documentation examples, it only builds the project and exits.

This is when running cargo from nightly.
```
mk@maero ~/Projects/SoundCloud % cargo test --doc
mk@maero ~/Projects/SoundCloud % 
```

**Comment by mkroman** [comment]
I've submitted a pull request. #2685

**Comment by huonw** [comment]
Fixed by #2685.

**Comment by dianpopa** [comment]
Hi, 

I am able to replicate this problem with `rustdoc 1.27.2 (58cc626de 2018-07-18)`
and `rustc 1.27.2 (58cc626de 2018-07-18)`. Running `cargo test --doc` exits without running documentation examples even though the folders that I am trying to run the command for are library crate-types. Any suggestions?

**Comment by sorenmortensen** [comment]
+1 on 1.27.0 and 1.27.2.

**Comment by ehuss** [comment]
@dianpopa Can you post an example package that demonstrates the problem?

@sorenmortensen `cargo test` should run the doc tests.  Is cargo not even printing `Doc-tests my-package`?  Does it happen with a simple package with just a library?

**Comment by sorenmortensen** [comment]
@ehuss That's correct, it doesn't even print `Doc-tests foo`. I did just create a simple library package, though, and haven't been able to reproduce it yet. Will keep trying.

**Comment by sorenmortensen** [comment]
I believe I've found the culprit. It's because I've set the `crate-type` in `Cargo.toml`. I'm using `crate-type = ["staticlib"]`, but I can reproduce this also with `"dylib"`. Should I open a new issue, as this seems to be different?

[Here](https://github.com/sorenmortensen/doc-tests-example) is a minimal example that reproduces the problem.

**Comment by ehuss** [comment]
@sorenmortensen Thanks for the info.  Only `rlib`s support doctests.  It was an unintentional change to allow you to even try with `test --doc`.  I'll take care of that, thanks!

**Comment by dianpopa** [comment]
I discovered what was the problem for me. If I remove ```x86_64-unknown-linux-musl``` target from `.cargo/config` I am able to see the `Doc-tests` being run. I am compiling on an x86 machine. If I keep the musl target and I specifically run cargo test like ```cargo test --target x86_64-unknown-linux-gnu``` the `Doc-tests` are successfully run. I am guessing I stumbled upon https://github.com/rust-lang/cargo/issues/4471 which seems to be the expected  behavior. However, to me this is not exactly cross-compilation as I am on an x86 host and trying to obtain a musl enabled binary for the same type of architecture (i.e. x86). What is your opinion on this?
